### PR TITLE
Split OOM event into OOM and OOM Kill

### PR DESCRIPTION
**Pull request by vmarmol**

### CODE REVIEW

File: manager/manager.go [hunk 1]
```diff
@@ -876,21 +876,32 @@ func (self *manager) watchForNewOoms() error {
...
 			newEvent := &info.Event{
 				ContainerName: oomInstance.ContainerName,
 				Timestamp:     oomInstance.TimeOfDeath,
 				EventType:     info.EventOom,
+			}
+			err := self.eventHandler.AddEvent(newEvent)
+			if err != nil {
+				glog.Errorf("failed to add OOM event for %q: %v", oomInstance.ContainerName, err)
+			}
+			glog.V(2).Infof("Created an OOM event in container %q at %v", oomInstance.ContainerName, oomInstance.TimeOfDeath)
```
Comment (vishh) [inline manager/manager.go]: V(3) or V(4)?

**Comment by rjnagal** [comment]
So every OOM creates two events now? It seems a little confusing.

File: manager/manager.go (hunk 1 above)
Comment (vmarmol) [inline manager/manager.go]: Done

**Comment by vishh** [comment]
LGTM.

**Comment by vmarmol** [comment]
@rjnagal its meant to simplify and allow for watching of OOM kills and OOMs separately.

For example:
Container /a/b watches for OOM kills on itself and its children. Container /a OOMs and kills a process in /a/b/c. Currently, my notification won't fire, with the new interface it will.

It is true that not all OOMs lead to OOM kills (and we don't handle those today, we should).

**Comment by rjnagal** [comment]
LGTM

I understand the rationale, but I worry that people not familiar with OOM working will get confused.
Let's get it in for now as your use-case is legitimate and needs to be covered. The API solution might be a more sophisticated spec when registering for watches.

**Pull request merged**